src/containers/BuySellCrypto/BuyCrypto/BuyCrypto.js

```diff
@@ -471,6 +460,37 @@ class BuyCrypto extends Component {
               }
               </FormValidationMessage>
             </View>
+      </View>
+    )
+  }
+
+  render() {
+    return (
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+        <View style={styles.root}>
+          <Spinner
+            visible={this.state.loadingOverlay || this.state.addingCoin || this.props.inProgress || this.props.isExchangeRatesFetching}
+            textContent={
+              this.state.addingCoin ?
+                "Adding coin..."
+                :
+                "Loading..."
+            }
+            textStyle={{color: "#FFF"}}
+          />
+          <ScrollView
+            contentContainerStyle={{height: "100%", alignItems: "center", justifyContent: "center"}}
+            refreshControl={
+              <RefreshControl
+                refreshing={this.state.loading}
+                onRefresh={this.forceUpdate}
+              />
+          }>
+            
+          {this.props.buy && this.renderFromCurr()}
+          {this.props.buy && this.renderToCurr()}
+          {!this.props.buy && this.renderToCurr()}
```

Comment:
<Dropdown
  data={this.props.buy ? SUPPORTED_FIAT_CURRENCIES : SUPPORTED_CRYPTOCURRENCIES}
/>

Reply:
switchToCurr = (coin) => {
    const coinIsAlreadyAddedToWallet = (this.props.activeCoinsForUser.find(coinObj => {
      return SUPPORTED_CRYPTOCURRENCIES.map(x => x.value).includes(coinObj.id)
    }))
    if (this.props.buy && !coinIsAlreadyAddedToWallet) {
      this.canAddCoin(coin)
      .then((res) => {
        this.setState({loadingOverlay: true, addingCoin: true}, () => {
          if (res) {
            this.handleAddCoin(coin)
            .then((res) => {
              if (res) {
                this.handleUpdates()
                .then(() => {
                  this.setState({toCurr: coin, loadingOverlay: false, addingCoin: false})
                })
              }
            })
          }
        })
      })
    } else {
      this.setState({toCurr: coin})
    }
  }